OpenRA.Game/Activities/Activity.cs

```diff
@@ -120,7 +120,8 @@ public Activity TickOuter(Actor self)
 				lastRun = Tick(self);
 
 			// Avoid a single tick delay if the childactivity was just queued.
-			if (ChildActivity != null && ChildActivity.State == ActivityState.Queued)
+			var ca = ChildActivity;
+			if (ca != null && ca.State == ActivityState.Queued)
```

Comment:
Are we sure that this is note done in the optimization? Ie have you checked the IL?

Reply:
I'm far from an expert at understanding IL, but this does appear to be evaluating `ChildActivity` twice on bleed:
```
// if ChildActivity != null
	IL_00c2:  ldarg.0 
	IL_00c3:  call instance class OpenRA.Activities.Activity class OpenRA.Activities.Activity::get_ChildActivity()
	IL_00c8:  brfalse.s IL_0101

// if ChildActivity.State == ActivityState.Queued
	IL_00ca:  ldarg.0 
	IL_00cb:  call instance class OpenRA.Activities.Activity class OpenRA.Activities.Activity::get_ChildActivity()
	IL_00d0:  callvirt instance valuetype OpenRA.Activities.ActivityState class OpenRA.Activities.Activity::get_State()
	IL_00d5:  brtrue.s IL_0101

// if ChildHasPriority
	IL_00d7:  ldarg.0 
	IL_00d8:  call instance bool class OpenRA.Activities.Activity::get_ChildHasPriority()
	IL_00dd:  brfalse.s IL_00f9
```